ietf/sync/rfceditor.py

```diff
@@ -504,56 +556,96 @@ def parse_relation_list(l):
             return res
 
         for x in parse_relation_list(obsoletes):
-            if not RelatedDocument.objects.filter(source=doc, target=x, relationship=relationship_obsoletes):
-                r = RelatedDocument.objects.create(source=doc, target=x, relationship=relationship_obsoletes)
-                changes.append("created %s relation between %s and %s" % (r.relationship.name.lower(), prettify_std_name(r.source.name), prettify_std_name(r.target.name)))
+            if not RelatedDocument.objects.filter(
+                source=doc, target=x, relationship=relationship_obsoletes
+            ):
+                r = RelatedDocument.objects.create(
+                    source=doc, target=x, relationship=relationship_obsoletes
+                )
+                rfc_changes.append(
+                    "created {rel_name} relation between {src_name} and {tgt_name}".format(
+                        rel_name=r.relationship.name.lower(),
+                        src_name=prettify_std_name(r.source.name),
+                        tgt_name=prettify_std_name(r.target.name),
+                    )
+                )
 
         for x in parse_relation_list(updates):
-            if not RelatedDocument.objects.filter(source=doc, target=x, relationship=relationship_updates):
-                r = RelatedDocument.objects.create(source=doc, target=x, relationship=relationship_updates)
-                changes.append("created %s relation between %s and %s" % (r.relationship.name.lower(), prettify_std_name(r.source.name), prettify_std_name(r.target.name)))
+            if not RelatedDocument.objects.filter(
+                source=doc, target=x, relationship=relationship_updates
+            ):
+                r = RelatedDocument.objects.create(
+                    source=doc, target=x, relationship=relationship_updates
+                )
+                rfc_changes.append(
+                    "created {rel_name} relation between {src_name} and {tgt_name}".format(
+                        rel_name=r.relationship.name.lower(),
+                        src_name=prettify_std_name(r.source.name),
+                        tgt_name=prettify_std_name(r.target.name),
+                    )
+                )
 
         if also:
             for a in also:
                 a = a.lower()
                 if not DocAlias.objects.filter(name=a):
                     DocAlias.objects.create(name=a).docs.add(doc)
-                    changes.append("created alias %s" % prettify_std_name(a))
+                    rfc_changes.append(f"created alias {prettify_std_name(a)}")
 
-        doc_errata = errata.get('RFC%04d'%rfc_number, [])
-        all_rejected = doc_errata and all( er['errata_status_code']=='Rejected' for er in doc_errata )
+        doc_errata = errata.get("RFC%04d" % rfc_number, [])  # rfc10k problem here
```

Comment:
Lets get rid of the %04 here now?

Reply:
do errata searches actually take leading zeros?

Reply:
I can make this handle leading zeroes or not - that'll avoid having to closely coordinate the change.